dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorChannel.java

```diff
@@ -129,11 +129,12 @@ LimitedChannel getCurrentChannel() {
                 .map(future -> DialogueFutures.addDirectCallback(future, new FutureCallback<Response>() {
                     @Override
                     public void onSuccess(Response response) {
-                        if (Responses.isQosStatus(response) || Responses.isServerError(response)) {
+                        // We *only* switch nodes on server *errors*, NOT QoS responses to support transactional
+                        // workflows where it is important for a large number of requests to all land on the same node,
+                        // even if a couple of them get rate limited in the middle.
+                        if (Responses.isServerError(response)) {
```

Comment:
I think it makes sense to remain on the same node in the face of a 429, but it seems wrong to stay pinned after a 503